File: sedmica 3/FIT_Api_Examples/FIT_Api_Example/Endpoints/Student/Pretraga/StudentPretragaEndpoint.cs
using FIT_Api_Example.Data;
using FIT_Api_Example.Helper;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;

namespace FIT_Api_Example.Endpoints.Student.Pretraga
{
    [Route("student-pretraga")]
    public class StudentPretragaEndpoint: MyBaseEndpoint<StudentPretragaRequest,  StudentPretragaResponse>
    {
        private readonly ApplicationDbContext _applicationDbContext;

        public StudentPretragaEndpoint(ApplicationDbContext applicationDbContext)
        {
            _applicationDbContext = applicationDbContext;
        }

        [HttpGet]
        public override async Task<StudentPretragaResponse> Obradi([FromQuery]StudentPretragaRequest request)
        {
            var student = await _applicationDbContext.Student.Where(x =>
                    request.Pretraga == null || 
                    (x.Ime + " " + x.Prezime).StartsWith(request.Pretraga) ||
                    (x.Prezime + " " + x.Ime).StartsWith(request.Pretraga) 
                    )
                .OrderByDescending(x=>x.ID)
                .Select(x=>new StudentPretragaResponseStudent()
                {
                    ID = x.ID,
                    DatumRodjenja = x.DatumRodjenja,
                    Ime = x.Ime,
                    Prezime =  x.Prezime,
                    KorisnickoIme = x.KorisnickoIme,
                    OpstinaRodjenjaDrzava = x.OpstinaRodjenja.drzava.Naziv,
                    OpstinaRodjenjaNaziv = x.OpstinaRodjenja.description,
                    SlikaKorisnika = x.SlikaKorisnika
                })
                .ToListAsync();

            return new StudentPretragaResponse
            {
                Studenti = student
            };
        }
    }
}

File: sedmica 3/front-end/src/app/endpoints/student-endpoints/student-pretraga.endpoint.ts
import {MyBaseEndpoint} from "../../helpers/my-base-endpoint";
import {MojConfig} from "../../moj-config";
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Injectable} from "@angular/core";
@Injectable({providedIn: 'root'})
export class StudentPretragaEndpoint implements  MyBaseEndpoint<string, StudentPretragaResponse>{
  constructor(public httpClient:HttpClient) { }

  obradi(naziv: string): Observable<StudentPretragaResponse> {
    let url=MojConfig.adresa_servera+`/student/pretraga?Pretraga=${naziv}`;
      return this.httpClient.get<StudentPretragaResponse>(url);
    }
}

export interface StudentPretragaResponse {
  studenti: StudentPretragaResponseStudent[];
}

export interface StudentPretragaResponseStudent {
  slikaKorisnika: string;
  id: number
  ime: string
  prezime: string
  opstinaRodjenjaNaziv: string
  opstinaRodjenjaDrzava: string
  datumRodjenja: string
  korisnickoIme: string
}

